Releasenotes/2020.2.4 asteria (#71)
kasenvr/vircadia-docs-sphinx: docs/source/release-notes/2020-2-4-Asteria.md

# Alpha 2020.2.4 Asteria Release Notes

This version of Vircadia is an update to 2020.2.3. It is compatible with the 2020.2.0-Asteria protocol. This version was released on August 6th, 2020.

## Interface (Codename Athena)

The numbers at the end of each item are the PR numbers in the Project Athena [repo](https://github.com/kasenvr/project-athena/).

#### General

* API: `Window.getUserAgent` has been added to return the Interface's user agent. (#520)

> * App: The Inventory v2.0.2 has been merged as an unpublished script, to be deployed in the More app. (#521)

Reviewer: ```suggestion
* App: The Inventory app v2.0.2 has been merged as an unpublished script, to be deployed in the More app. (#521)
```

* Graphics: Fix bug where FBX models will only show vertex colors if more than one color is given. (#545)
* Avatar: Fix bug where Interface lags extensively when moving after startup due to a physics issue. (#547)
* UI: The graphics settings dialog now allows you to set the resolution scale to 2.0, up from 1.0. (#550)
* Graphics: Fix bug where `renderWithZones` doesn't work after reloading content. (#561)
* Graphics: Fix bug where `renderWithZones` would not work when loading content from a JSON file. (#564)
* Application: Fix bug where the --url command line parameter would not work under some circumstances. (#574)
* App: The Chat app has a few updates. (#579)
    * A notification has been added for when people enter or leave the domain.
    * The main Chat window will open when you click on a chat message notification.
* Security: You can now sign-in to OAuth2 enabled domains. (Alpha Feature) (#591)
* Graphics: The default "Automatic" maximum texture memory setting has been increased from 1GB to 2GB. (#594)
* UI: The graphics settings dialog now has a button in the toolbar for clear and easy access. (#595)

#### Technicals

> * Docs: Packaging docs are more accurate now to the requirements on Windows. (#537)

Reviewer: ```suggestion
* Docs: Packaging docs are more accurate now with respect to the requirements on Windows. (#537)
```

> * UX: The NSIS installer no longer defaults the option for the sandbox (server-console) to start with Windows as true. (#596)

Reviewer: ```suggestion
* UX: The NSIS installer no longer defaults the options for the Sandbox (server-console) to start after install or start with Windows as true. (#596)
```

## Domain Server

The numbers at the end of each item are the PR numbers in the Project Athena [repo](https://github.com/kasenvr/project-athena/).

#### General

* Security: Domains can now accept user logins via groups based on WordPress roles via OAuth2. (Alpha Feature) (#591)

## Metaverse Server (Codename Apollo)

> The [metaverse server](https://github.com/kasenvr/project-apollo) and [metaverse server interface](https://github.com/kasenvr/project-apollo-dashboard) are progressing well. The metaverse server is currently being rewritten in TypeScript with Node.JS in order to improve its capabilities for the long term. As of now (8/6/2020), it is at feature parity with its original C# implementation.

Reviewer: ```suggestion
The [metaverse server](https://github.com/kasenvr/project-apollo) and [metaverse server interface](https://github.com/kasenvr/project-apollo-dashboard) are progressing well. The metaverse server is currently being rewritten in TypeScript with Node.JS in order to improve its capabilities for the long term. As of now (8/6/2020) it is at feature parity with its original C# implementation.
```

## Vircadia Launcher (Codename Pantheon)

> [Vircadia's launcher](https://github.com/kasenvr/pantheon-launcher) is at version 0.0.1u. The launcher now runs the Interface as a separate process by default. Because of this, it now has heavily improved stability when Interface is launched with it. The drag and drop feature as well as crash reporting will now also work correctly.

Reviewer: ```suggestion
[Vircadia's launcher](https://github.com/kasenvr/pantheon-launcher) is at version 0.0.1u. The launcher now runs the Interface as a separate process by default. Because of this, it now has greatly improved stability when Interface is launched with it. The drag and drop feature as well as crash reporting now also work correctly.
```
